ui/store/actions.js

```
@@ -2886,6 +2886,11 @@ export async function setSmartTransactionsOptInStatus(optInState) {
   await promisifiedBackground.setSmartTransactionsOptInStatus(optInState);
 }
 
+const isPersistentError = (e) => {
+  const persistentError = "Fetch failed with status '5";
```

I'm not sure that this error message would happen for all failed 5xx calls. Can we look for an HTTP response status code instead?

---

This error message is returned from the custom fetch function for all metamask calls so I think it should be sufficient. I agree that the status code is a more standard approach but I don't think the status code is available this far removed from the actual fetch call. The custom fetch swallows the error message and passes this custom error message which this line parses essentially.